daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package speedy
+
+import java.lang.System
+import java.util.ArrayList
+
+/** Class for profiling information collected by Speedy. We use [[AnyRef]] for
+  * the labels to avoid any conversions into a common label format at runtime
+  * since this might skew the profile. Instead, we convert the labels to strings
+  * when we write out the profile.
+  */
+final class Profile {
+  import Profile._
+  private val start: Long = System.nanoTime()
+  private val events: ArrayList[Event] = new ArrayList()
+  private var name: String = "DAML Engine profile"
+
+  def addOpenEvent(label: AnyRef) = {
+    val time = System.nanoTime()
+    events.add(Event(true, label, time))
+  }
+
+  def addCloseEvent(label: AnyRef) = {
+    val time = System.nanoTime()
+    events.add(Event(false, label, time))
+  }
+
+  def setName(name: String) = {
+    this.name = name
+  }
```

Comment:
Scala style for this pattern is to _not_ have a `setX` function, but for `.name = ...` to be the syntax used by callers. This can be done by making the `var` public or by implementing the `name_=` method.

Reply:
Good to know. I'll change it accordingly.